src/VS.Mefx/ConfigCreator.cs

```diff
@@ -0,0 +1,491 @@
+            if (isCacheFile)
+            {
+                this.CachePaths.Add(fullPath);
+            }
+            else
+            {
+                this.AssemblyPaths.Add(fullPath);
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Method to add valid files from the current folder and its subfolders to the list of paths.
+        /// </summary>
+        /// <param name="currentPath">The complete path to the folder we want to add files from.</param>
+        private void SearchFolder(string currentPath)
+        {
+            DirectoryInfo currentDir = new DirectoryInfo(currentPath);
+            var files = currentDir.EnumerateFiles();
+            foreach (var file in files)
+            {
+                string name = file.Name;
+                this.AddFile(currentPath, name);
+            }
+
+            IEnumerable<DirectoryInfo> subFolders = currentDir.EnumerateDirectories();
+            if (subFolders.Count() > 0)
+            {
+                foreach (DirectoryInfo subFolder in subFolders)
+                {
+                    this.SearchFolder(subFolder.FullName);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Method to read the input parts stored in cache files and add them to the existing Catalog.
+        /// </summary>
+        /// <param name="discovery">Part Discovery object to use when discovering parts in assembly.</param>
+        private async Task ReadCacheFiles(PartDiscovery discovery)
+        {
+            foreach (string filePath in this.CachePaths)
+            {
+                try
+                {
+                    using (FileStream inputStream = File.Open(filePath, FileMode.Open, FileAccess.Read, FileShare.Read))
+                    {
+                        CachedCatalog catalogReader = new CachedCatalog();
+                        ComposableCatalog cacheParts = await catalogReader.LoadAsync(inputStream, discovery.Resolver);
+
+                        if (this.Catalog == null)
+                        {
+                            this.Catalog = cacheParts;
+                        }
+                        else
+                        {
+                            this.Catalog = this.Catalog.AddCatalog(cacheParts);
+                        }
+                    }
+                }
+                catch (Exception error)
+                {
+                    string cacheReadError = string.Format(
+                        CultureInfo.CurrentCulture,
+                        Strings.ErrorMessage,
+                        error.Message);
+                    this.Options.ErrorWriter.WriteLine(cacheReadError);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Method to store the parts read from the input files into a cache for future use.
+        /// </summary>
+        private async Task SaveToCache()
```

Comment:
All methods returning awaitable types should use the `Async` suffix in its name.